h2/src/main/org/h2/expression/SearchedCase.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright 2004-2020 H2 Group. Multiple-Licensed under the MPL 2.0,
+ * and the EPL 1.0 (https://h2database.com/html/license.html).
+ * Initial Developer: H2 Group
+ */
+package org.h2.expression;
+
+import java.util.Arrays;
+
+import org.h2.engine.Session;
+import org.h2.value.TypeInfo;
+import org.h2.value.Value;
+import org.h2.value.ValueNull;
+
+/**
+ * A searched case.
+ */
+public class SearchedCase extends OperationN {
+
+    private int argsCount;
```

Comment:
if you do this, you need to override getSubexpressionCount. But why not just use ArrayList, rather than replicating the functionality?
Looking at this and the other places we're now using OperationN, perhaps OperationN should not be resposible for argument storage at all, OperationN can implement all its logic in terms of getSubexpression() and getSubexpressionCount().

Reply:
I don't know why `Function` uses an own list, I just preserved it as it.

But I agree, its logic can be moved to `OperationN`.

Reply:
`Function` uses own special logic in `addParameter()` to throw the exception a little big earlier. But I think there is no big difference when the exception will be thrown.